src/jit/codegenxarch.cpp

```diff
@@ -1813,162 +1813,20 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
                     assert(structDesc.eightByteCount == CLR_SYSTEMV_MAX_EIGHTBYTES_COUNT_TO_PASS_IN_REGISTERS);
 
                     regNumber retReg0 = REG_NA;
-                    emitAttr size0 = EA_UNKNOWN;
-                    unsigned offset0 = structDesc.eightByteOffsets[0];
+                    unsigned __int8 offset0 = 0;
```

Comment:
@sivarv I agree. We should not be combining multiple issues into a single PR. Let's try to focus on the changes and issues being specifically addressed by this PR, and open up additional issues as needed.